compiler/rustc_mir_transform/src/sroa.rs

```diff
@@ -31,8 +31,9 @@ impl<'tcx> crate::MirPass<'tcx> for ScalarReplacementOfAggregates {
         let mut excluded = excluded_locals(body);
         let typing_env = body.typing_env(tcx);
         loop {
+            add_type_based_exclusions(tcx, &mut excluded, body);
```

Comment:
Does this need to happen in the loop? The set of excluded locals based off of *type* exclusions shouldn't change, right?